cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/PullDocIdsFilter.java

```diff
@@ -0,0 +1,42 @@
+package com.cloudant.sync.replication;
+
+import com.cloudant.sync.internal.util.JSONUtils;
+import com.cloudant.sync.internal.util.Misc;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * <p>Sets the list of doc IDs to use as filtering criteria when a pull replication calls the
+ * source database's {@code _changes} feed.
+ * </p>
+ * Note: Doc IDs filtering are supported only when replicating against a CouchDB 2.x compliant database
+ *
+ * @see
+ * <a target="_blank" href="https://console.bluemix.net/docs/services/Cloudant/api/database.html#get-changes">See doc_ids filtering</a>
+ */
+public class PullDocIdsFilter extends PullFilter {
+
+    public static final String DOC_IDS_FILTER_NAME = "_doc_ids";
+
+    public PullDocIdsFilter(List<String> docIds) {
+        super(DOC_IDS_FILTER_NAME);
+
+        Misc.checkState(docIds != null && !docIds.isEmpty(),
+                "Doc Ids should not be empty in a Doc Ids Filter");
+
+        List<String> internalDocIds = new ArrayList<String>(docIds);
+        Collections.sort(internalDocIds);
```

Comment:
Is there any reason for sorting doc ids?

Reply:
As we need to build a constant replicationId the sorting allows to keep the replication id constant even if the aplication provides de ids in different order. 

I was not sure if it could be convenient to remove duplicated Ids from the list.